tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsContextPropagator.cs

```diff
@@ -47,4 +49,46 @@ public void Inject<TCarrier>(PathwayContext context, TCarrier headers)
 
         return bytes is { } ? PathwayContextEncoder.Decode(bytes) : null;
     }
+
+    /// <summary>
+    /// Propagates the specified context by adding new headers to a <see cref="IHeadersCollection"/>.
+    /// This locks the sampling priority for <paramref name="context"/>.
+    /// </summary>
+    /// <param name="context">A <see cref="PathwayContext"/> value that will be propagated into <paramref name="headers"/>.</param>
+    /// <param name="headers">A <see cref="IHeadersCollection"/> to add new headers to.</param>
+    /// <typeparam name="TCarrier">Type of header collection</typeparam>
+    public void InjectAsBase64String<TCarrier>(PathwayContext context, TCarrier headers)
+        where TCarrier : IHeadersCollection
+    {
+        if (headers is null) { ThrowHelper.ThrowArgumentNullException(nameof(headers)); }
+
+        headers.Add(DataStreamsPropagationHeaders.PropagationKey, Convert.ToBase64String(PathwayContextEncoder.Encode(context)));
```

Comment:
The main limitation I had when trying to set the pathway context in the message headers is that I can't directly pass in a byte array, so we have to pass in a string of some sort. So the question becomes what encoding should we use. base64 made sense to me since we can easily encode/decode the 8-bit values this way, but I'm following up with the DSM team to confirm.